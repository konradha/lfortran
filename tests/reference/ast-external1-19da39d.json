{
    "basename": "ast-external1-19da39d",
    "cmd": "lfortran --indent --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/external1.f90",
    "infile_hash": "d1e5f47338a407a9c31e7dc02f301593a20de1be72b878859b097486",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-external1-19da39d.stdout",
    "stdout_hash": "4788df02ac9fe3e48d85f51b0c04f895e0a5745130b7c801204eb71d",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}