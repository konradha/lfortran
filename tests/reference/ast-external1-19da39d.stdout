(TranslationUnit 
    [(Program 
        main 
        (TriviaNode 
            [] 
            [(EndOfLine)
            (EndOfLine)]
        ) 
        [] 
        [] 
        [(Declaration 
            () 
            [(SimpleAttribute 
                AttrExternal
            )] 
            [(sub1 
            [] 
            [] 
            () 
            None 
            ())] 
            (TriviaNode 
                [] 
                [(EOLComment 
                    "! external statement"
                )]
            )
        )
        (Declaration 
            () 
            [(SimpleAttribute 
                AttrExternal
            )] 
            [(fun1 
            [] 
            [] 
            () 
            None 
            ())] 
            (TriviaNode 
                [] 
                [(EOLComment 
                    "! external attribute"
                )]
            )
        )
        (Declaration 
            (AttrType 
                TypeInteger 
                [] 
                () 
                None
            ) 
            [(SimpleAttribute 
                AttrExternal
            )] 
            [(fun2 
            [] 
            [] 
            () 
            None 
            ())] 
            ()
        )] 
        [(SubroutineCall 
            0 
            g 
            [] 
            [(() 
            sub1 
            ())
            (() 
            fun1 
            ())] 
            [] 
            ()
        )] 
        []
    )
    (Subroutine 
        g 
        [(proc)
        (fun)] 
        [] 
        () 
        (TriviaNode 
            [] 
            [(EndOfLine)
            (EndOfLine)]
        ) 
        [] 
        [] 
        [] 
        [(Declaration 
            (AttrType 
                TypeInteger 
                [] 
                () 
                None
            ) 
            [] 
            [(fun 
            [] 
            [] 
            () 
            None 
            ())
            (i 
            [] 
            [] 
            () 
            None 
            ())] 
            ()
        )] 
        [(= 
            0 
            i 
            (FuncCallOrArray 
                fun 
                [] 
                [] 
                [] 
                []
            ) 
            ()
        )
        (SubroutineCall 
            0 
            proc 
            [] 
            [(() 
            i 
            ())] 
            [] 
            ()
        )] 
        []
    )
    (Subroutine 
        sub1 
        [(i)] 
        [] 
        () 
        (TriviaNode 
            [] 
            [(EndOfLine)
            (EndOfLine)]
        ) 
        [] 
        [] 
        [] 
        [(Declaration 
            (AttrType 
                TypeInteger 
                [] 
                () 
                None
            ) 
            [] 
            [(i 
            [] 
            [] 
            () 
            None 
            ())] 
            ()
        )] 
        [(Write 
            0 
            [(())
            (())] 
            [] 
            [i] 
            ()
        )] 
        []
    )
    (Function 
        fun1 
        [] 
        [(AttrType 
            TypeInteger 
            [] 
            () 
            None
        )] 
        () 
        () 
        (TriviaNode 
            [] 
            [(EndOfLine)
            (EndOfLine)
            (Comment 
                "! TODO the below should work fine the moment LFortran implements statement functions"
            )
            (Comment 
                "!integer function fun2()"
            )
            (Comment 
                "!    integer a"
            )
            (Comment 
                "!    a = 100"
            )
            (Comment 
                "!    fun2 = a + 1"
            )
            (Comment 
                "!end function"
            )]
        ) 
        [] 
        [] 
        [] 
        [(Declaration 
            (AttrType 
                TypeInteger 
                [] 
                () 
                None
            ) 
            [] 
            [(a 
            [] 
            [] 
            () 
            None 
            ())] 
            ()
        )] 
        [(= 
            0 
            a 
            100 
            ()
        )
        (= 
            0 
            a 
            (+ a 1) 
            ()
        )] 
        []
    )]
)
